tests/unit/sagemaker/jumpstart/curated_hub/test_utils.py

```diff
@@ -168,3 +174,110 @@ def test_create_hub_bucket_if_it_does_not_exist():
 
     mock_sagemaker_session.boto_session.resource("s3").create_bucketassert_called_once()
     assert created_hub_bucket_name == bucket_name
+
+@patch("sagemaker.jumpstart.utils.verify_model_region_and_return_specs")
+def test_find_tags_for_jumpstart_model_version(mock_spec_util):
+    mock_sagemaker_session = Mock()
+    mock_specs = Mock()
+    mock_specs.deprecated = True
+    mock_specs.inference_vulnerable = True
+    mock_specs.training_vulnerable = True
+    mock_spec_util.return_value = mock_specs
+
+    tags = utils.find_jumpstart_tags_for_model_version(
+        model_id="test",
+        version="test",
+        region="test",
+        session=mock_sagemaker_session
+    )
+
+    mock_spec_util.assert_called_once_with(
+        model_id="test",
+        version="test",
+        region="test",
+        scope=JumpStartScriptScope.INFERENCE,
+        tolerate_vulnerable_model = True,
+        tolerate_deprecated_model = True,
+        sagemaker_session=mock_sagemaker_session,
+    )
+
+    assert tags == [CuratedHubTagName.DEPRECATED_VERSIONS_TAG, CuratedHubTagName.INFERENCE_VULNERABLE_VERSIONS_TAG, CuratedHubTagName.TRAINING_VULNERABLE_VERSIONS_TAG]
+
+@patch("sagemaker.jumpstart.utils.verify_model_region_and_return_specs")
+def test_find_tags_for_jumpstart_model_version_some_false(mock_spec_util):
+    mock_sagemaker_session = Mock()
+    mock_specs = Mock()
+    mock_specs.deprecated = True
+    mock_specs.inference_vulnerable = False
+    mock_specs.training_vulnerable = False
+    mock_spec_util.return_value = mock_specs
+
+    tags = utils.find_jumpstart_tags_for_model_version(
+        model_id="test",
+        version="test",
+        region="test",
+        session=mock_sagemaker_session
+    )
+
+    mock_spec_util.assert_called_once_with(
+        model_id="test",
+        version="test",
+        region="test",
+        scope=JumpStartScriptScope.INFERENCE,
+        tolerate_vulnerable_model = True,
+        tolerate_deprecated_model = True,
+        sagemaker_session=mock_sagemaker_session,
+    )
+
+    assert tags == [CuratedHubTagName.DEPRECATED_VERSIONS_TAG]
+
+@patch("sagemaker.jumpstart.utils.verify_model_region_and_return_specs")
+def test_find_all_tags_for_jumpstart_model(mock_spec_util):
```

Comment:
Can you please add a test for:

1. When a version is either vulnerable OR deprecated, not both
2. When multiple version are vulnerable, but one is deprecated
3. When one version is vulnerable, and one is deprecated

Reply:
The tests for the `find_jumpstart_tags_for_model_version` do this, no?